dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/AbstractDatadogSparkListener.java

```
@@ -1256,6 +1258,48 @@ private static String getStreamingBatchKey(String queryId, String batchId) {
     return queryId + "." + batchId;
   }
 
+  @SuppressForbidden
+  private static Object getStreamExecutionQueryIdKey() {
+    try {
+      Class<?> cls = Class.forName("org.apache.spark.sql.execution.streaming.StreamExecution");
+      try {
+        Field f = cls.getDeclaredField("QUERY_ID_KEY");
+        f.setAccessible(true);
+        return f.get(null);
+      } catch (NoSuchFieldException e) {
+        try {
+          Method m = cls.getDeclaredMethod("QUERY_ID_KEY");
+          m.setAccessible(true);
+          return m.invoke(null);
+        } catch (Throwable ignored) {
+        }
+      }
+    } catch (Throwable ignored) {
+    }
+    return null;
+  }
+
+  @SuppressForbidden
+  private static Object getMicroBatchExecutionBatchIdKey() {
```

You can give it a try. In OL connector, we're using a bit more complex version:
```
  private static Object getMicroBatchExecutionBatchIdKey() {
    // make it constant
    String microBatchExecutionClass = "org.apache.spark.sql.execution.streaming.MicroBatchExecution";

    // Checking if a Class Exists
    if (!classIsLoadable(microBatchExecutionClass)) {
      return null;
    }

    try {
      // Getting the Scala Object Instance
      Class<?> cls = Class.forName(microBatchExecutionClass);
      Object module = cls.getField("MODULE$").get(null);

      // Accessing Members via Reflection
      java.lang.reflect.Method method = cls.getMethod("BATCH_ID_KEY");
      return (String)method.invoke(module);
    } catch (ClassNotFoundException | NoSuchFieldException | IllegalAccessException | NoSuchMethodException |
             InvocationTargetException e) {
      // ignore
    }
    return null;
  }
```

I think it's necessary here as well. It's 4 steps:
 * check class existince
 * get Scala object by getting MODULE field on Scala class
 * get BATCH_ID_KEY method as Scala objects generate methods for each fields
 * call BATCH_ID_KEY method

---

The code works well when `MicroBatchExecution` class ain't present and this fixes customer's issue. 
But, I am not sure if the behaviour is going to be the same for cases when `MicroBatchExecution` will be present.

Perhaps you can re-throw the exceptions caught on testing branch and see if the tests, particularly `datadog.trace.instrumentation.spark.AbstractSparkStructuredStreamingTest` test is still passing. The test seem to check the existence of `streaming_sql_batchId` in the span but not its value, so the test may not detect issues when errors are silenced. But the test should definitely fail if an exception gets thrown.